src/components/side-panel/interaction-panel/tooltip-config.js

```diff
@@ -18,40 +18,301 @@
+                    <FormattedMessage id={'fieldSelector.formatting'} />
+                  )}
+                </span>
+              </Tooltip>
+            </ChickletAddon>
+            {show && (
+              <StyledPopover>
+                <DropdownList
+                  options={getFormatLabels()}
+                  selectionIndex={selectionIndex}
+                  onOptionSelected={(result, e) => {
+                    e.stopPropagation();
+                    setShow(false);
+                    setSelectionIndex(getFormatLabels().indexOf(result));
+                    const oldFieldsToShow = config.fieldsToShow[dataId];
+                    const fieldsToShow = oldFieldsToShow.map(field => {
+                      const name = field.name || field;
+                      return name === getItemName(item)
+                        ? {
+                            name,
+                            format: result
+                          }
+                        : field;
+                    });
+                    const newConfig = {
+                      ...config,
+                      fieldsToShow: {
+                        ...config.fieldsToShow,
+                        [dataId]: fieldsToShow
+                      }
+                    };
+                    onChange(newConfig);
+                  }}
+                />
+              </StyledPopover>
+            )}
+          </ChickletAddonWrapper>
+        );
+      };
+    };
+
+    render() {
+      const {config, datasets, onChange} = this.props;
+      return (
+        <TooltipConfigWrapper>
+          {Object.keys(config.fieldsToShow).map(dataId => (
+            <SidePanelSection key={dataId}>
+              <SBFlexboxNoMargin>
+                <DatasetTag dataset={datasets[dataId]} />
+                {Boolean(config.fieldsToShow[dataId].length) && (
+                  <ButtonWrapper>
+                    <Button
+                      className="clear-all"
+                      onClick={() => {
+                        const newConfig = {
+                          ...config,
+                          fieldsToShow: {
+                            ...config.fieldsToShow,
+                            [dataId]: []
+                          }
+                        };
+                        onChange(newConfig);
+                      }}
+                      width="48px"
+                      secondary
+                    >
+                      <FormattedMessage id="fieldSelector.clearAll" />
+                    </Button>
+                  </ButtonWrapper>
+                )}
+              </SBFlexboxNoMargin>
+              <FieldSelector
+                fields={datasets[dataId].fields}
+                value={config.fieldsToShow[dataId]}
+                onSelect={fieldsToShow => {
+                  const oldFieldsToShow = config.fieldsToShow[dataId];
+                  const newFieldsToShow = fieldsToShow.map(field => {
```

Comment:
this will result in an array contains undefined. Why do you have to this?

Reply:
Can you explain how. Had a second look into it...